static/js/reload.js

```diff
@@ -99,11 +99,34 @@ function preserve_state(send_after_reload, save_pointer, save_narrow, save_compo
     // TODO: Remove the now-unnecessary URL-encoding logic above and
     // just pass the actual data structures through local storage.
     const token = util.random_int(0, 1024 * 1024 * 1024 * 1024);
-
-    ls.set("reload:" + token, url);
+    const metadata = {
+        url,
+        date: Date.now(),
+    };
+    ls.set("reload:" + token, metadata);
     window.location.replace("#reload:" + token);
 }
 
+function delete_stale_tokens(ls) {
+    // The time between reload token generation and use should usually be
+    // fewer than 30 seconds, but we keep tokens around for a week just in case
+    // (e.g. a tab could fail to load and be refreshed a while later).
+    const now = Date.now();
+    const milliseconds_in_a_day = 1000 * 60 * 60 * 24;
+    ls.removeDataRegexWithCondition("reload:\\d+", (metadata) => {
+        // TODO: In Sep 2022, the metadata was changed from a string to a map
+        // containing the string and a date. For now we'll delete all tokens
+        // that only contain the url. Once this has been shipped for a bit,
+        // we can remove this early return.
+        if (!metadata.date) {
+            return true;
+        }
+        const timedelta = now - metadata.date;
+        const days_since_token_creation = timedelta / milliseconds_in_a_day;
+        return days_since_token_creation > 7;
+    });
+}
```

Comment:
I think it'd be good to name the condition function, export it, and then we can write simple tests for the actual logic here here.

Reply:
Great call, doing that now